PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfigLoader.java

```diff
@@ -0,0 +1,175 @@
+
+/**
+ * This class is used to load the rule configurations from a JSON file.
+ * The rule configurations are loaded once and stored in a map.
+ * The map contains the key-value pairs where the key is the rule name and the value is the RuleConfig object.
+ * The RuleConfig object for a given key can be retrieved using the getRuleConfig method.
+ */
+public class RuleConfigLoader {
+
+    private static final RuleConfigLoader instance;
+    private final Map<String, RuleConfig> ruleConfigs;
+
+    private static final Logger LOGGER = Logger.getLogger(RuleConfigLoader.class.getName());
+
+    private static final String CONFIG_FILE_PATH = "/META-INF/ruleConfigs.json";
+
+    // Static initializer block to load the configurations once
+    static {
+        RuleConfigLoader instanceTemp;
+        try {
+            instanceTemp = new RuleConfigLoader(CONFIG_FILE_PATH);
+        } catch (FileNotFoundException e) {
+            instanceTemp = null;
+            LOGGER.log(Level.SEVERE, "Configuration file not found at path: " + CONFIG_FILE_PATH + ". Please ensure the file exists and is accessible. Error: " + e);
+        } catch (IOException e) {
+            instanceTemp = null;
+            LOGGER.log(Level.SEVERE, "IO error while reading configuration file from path: " + CONFIG_FILE_PATH + ". Please check file permissions and retry. Error: " + e);
+        }
+        instance = instanceTemp;
+    }
+
+    /**
+     * Constructor to load the rule configurations from the JSON file
+     *
+     * @param filePath - the path to the JSON file
+     * @throws IOException - if there is an error reading the file
+     */
+    private RuleConfigLoader(String filePath) throws IOException {
+        this.ruleConfigs = loadRuleConfigurations(filePath);
+    }
+
+    /**
+     * This method returns the instance of the CentralRuleConfigLoader
+     *
+     * @return CentralRuleConfigLoader instance
+     */
+    public static RuleConfigLoader getInstance() {
+        return instance;
+    }
+
+    /**
+     * This method returns the RuleConfig object for the given key
+     *
+     * @param key - the key to get the RuleConfig object
+     * @return RuleConfig object
+     */
+    RuleConfig getRuleConfig(String key) {
+        return ruleConfigs.get(key);
+    }
+
+    /**
+     * This method loads the rule configurations from the JSON file
+     *
+     * @param filePath - the path to the JSON file
+     * @return Map of RuleConfig objects
+     */
+    private Map<String, RuleConfig> loadRuleConfigurations(String filePath) {
+
+        // temporary map to store the RuleConfig objects that will then be returned to the final map
+        Map<String, RuleConfig> ruleConfigs = new HashMap<>();
+
+        // Open the input stream to the JSON file
+        try (InputStream is = RuleConfigLoader.class.getResourceAsStream(filePath);
+
+             // Create a JsonReader to read the JSON file
+             JsonReader reader = JsonProviders.createReader(is)) {
```

Comment:
Since this class implements Closeable, you want to make sure that you are disposing/closing the class when you're done using it.  In general, when I see "Reader" "Stream" "input", I'll check the class definition to see what interfaces it implements.

https://learn.microsoft.com/en-us/java/api/com.azure.json.jsonreader?view=azure-java-stable